lib/utils/fetchConversation.ts

```diff
@@ -1,108 +1,63 @@
-        'Content-Type': 'application/json',
-        'X-API-KEY': process.env.TRANSPOSE_API_KEY,
+  const response = await fetch('https://api.evm.storage/canvas_api.v1.CanvasApiService/Query', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      canvas_id: '9bd2632b-a0f4-40db-84c1-985c866e997e',
+      api_endpoint: 'get_conversation',
+      api_key: 'sim-api-e80d0a607c29a0d2',
+      query_parameters: {
+        chain_id: '1',
+        addr1: address1.toLowerCase(),
+        addr2: address2.toLowerCase(),
+        offset: (page * 10).toString(),
       },
-      body: JSON.stringify({
-        // Note: we select up to the most recent 100 messages whenever the data
-        // is stale or nonexistent.
-        // eslint-disable-next-line
-      sql: `SELECT t.block_number, t.from_address, t.to_address, t.value, raw.message, t.timestamp, t.transaction_hash FROM ( SELECT block_number, from_address, to_address, value, input, timestamp, transaction_hash FROM ethereum.transactions WHERE input IS NOT NULL AND ((from_address = \'{{address1}}\' AND to_address = \'{{address2}}\') OR (to_address = \'{{address1}}\' AND from_address = \'{{address2}}\')) ORDER BY block_number DESC ) AS t, LATERAL ( SELECT CONVERT_FROM( DECODE( SUBSTRING( REGEXP_REPLACE(t.input, \'00\', \'\', \'g\') FROM 3 ), \'hex\' ), \'LATIN1\' ) AS message ) AS raw WHERE LENGTH(raw.message) > 0 AND ( LENGTH(raw.message) - LENGTH(REGEXP_REPLACE(raw.message, '[\\u002D\\u0030-\\u0039\\u0041-\\u005A\\u005F\\u0061-\\u007A!@#$%^&*()_+<>]', '', 'g')) )::float / LENGTH(raw.message) >= 0.5 LIMIT 100 OFFSET {{offset}}`,
-        parameters: { address1: minAddress, address2: maxAddress, offset: page * 10 },
-      }),
-    });
-
-    if (!response || !response.ok) return [];
-
-    const json = await response.json();
-    if (!json || !json.results) return [];
-
-    const messages = json.results.map((tx: TransposeOnChainMessagesResponse) => ({
-      blockNumber: tx.block_number,
-      from: tx.from_address.toLowerCase(),
-      to: tx.to_address.toLowerCase(),
-      value: tx.value,
-      message: tx.message,
-      timestamp: new Date(tx.timestamp),
-      txHash: tx.transaction_hash.toLowerCase(),
-    }));
-
-    // Upsert into Supabase.
-    await supabase.from('chat_convo_last_updated').upsert({
-      address1: minAddress,
-      address2: maxAddress,
-      lastUpdated: new Date().toISOString(),
-    });
-    await supabase.from('chat_txs').upsert(messages);
-
-    return messages.slice(0, 100).reverse();
-  }
-
-  return data
-    .map(({ timestamp, ...rest }) => ({ timestamp: new Date(timestamp), ...rest }))
-    .reverse();
+    }),
+  });
+
+  const data = await response.json();
+  const rows = data.rows;
+  if (!rows) return [];
+
+  // data.columns will be ['block_number', 'from_addr', 'to_addr', 'value', 'message', 'msg_time', 'txn_hash']
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
I'm now just manually extracting the fields of `row` here but the proper way to do it is by parsing the following proto:

```
// This contains the query result from database, based on the connected query component and
// parameters in the request.
message QueryResponse {
  // The header for the result set, i.e. the column names.
  repeated string columns = 1;
  message Row {
    repeated Value value = 1;
  }
  // The result rows.
  repeated Row rows = 2;
  // If true, the result set was truncated due to exceeding the maximum allowed size.
  bool truncated = 3;
  // The time it took to execute the query in milliseconds.
  uint64 execution_time_ms = 4;
}

// If this proto is updated, please also update canvas.v1.ApiDataType enum.
message Value {
  oneof value {
    uint64 uint64 = 1;
    int64 int64 = 2;

    // These 2 fields are decimal strings like "43981".
    string uint256 = 3;
    string int256 = 4;

    // These 3 fields are 0x-prefixed hex strings like "0xabcd".
    string address = 5;
    string bytes32 = 6;
    string bytes = 7;

    string string = 8;
    bool bool = 9;

    uint32 uint32 = 10;
    int32 int32 = 11;
    float float32 = 12;
    double float64 = 13;

    // This field is a decimal string like "3.14".
    string big_decimal = 14;

    string json = 15;
  }
}
```

Reply:
You can also request a proto directly from our backend by using protobufs or even a compressed JSON via the `'Accept-Encoding': '*'` header in your JSON request